pyomo/network/port.py

```diff
@@ -0,0 +1,606 @@
+                       ("Name", "Value"), _line_generator)
+
+    def Equality(port, name, index_set):
+        """Arc Expansion procedure to generate simple equality constraints."""
+        # Iterate over every arc off this port. Since this function will
+        # be called for every port, we need to check if it already exists.
+        for arc in port.arcs(active=True):
+            Port._add_equality_constraint(arc, name, index_set)
+
+    def Extensive(port, name, index_set, write_var_sum=True):
+        """
+        Arc Expansion procedure for extensive variable properties.
+
+        This procedure is the rule to use when variable quantities should
+        be split for outlets and combined for inlets.
+
+        This will first go through every destination of the port and create
+        a new variable on the arc's expanded block of the same index as the
+        current variable being processed. It will also create a splitfrac
+        variable on the expanded block as well. Then it will generate
+        constraints for the new variable that relates it to its "parent"
+        variable by the split fraction. Following this, an indexed constraint
+        is written that states that the sum of all the new variables equals
+        the parent. However, if write_var_sum=False is passed, instead of
+        this last indexed constraint, a single constraint will be written
+        the states the sum of the split fractions equals 1.
+
+        Then, this procedure will go through every source of the port and
+        create a new variable (or use the same one as above), and then
+        write a constraint that states the sum of all the incoming new
+        variables must equal the parent variable.
+
+        Model simplifications:
+
+            If the port has a 1-to-1 connection on either side, it will not
+                create the new variables and instead write a simple
+                equality constraint for that side.
+
+            If the port has arcs on both sides but at least on of the sides
+                is 1-to-1, it will not write the bal constraint, since this
+                would be equivalent to the insum or outsum constraints.
+
+            If the outlet side is not 1-to-1 but there is only one outlet,
+                it will not create a splitfrac variable or write the split
+                constraint, but it will still write the outsum constraint
+                which will be a simple equality.
+
+            If the port only contains a single Extensive variable, the
+                splitfrac variables and the splitting constraints will
+                be skipped since they will be unnecessary.
+
+            Note: If split fractions are skipped, the write_var_sum=False
+                option is not allowed.
+        """
+        port_parent = port.parent_block()
+        out_vars = Port._Split(port, name, index_set, write_var_sum)
+        in_vars = Port._Combine(port, name, index_set)
```

Comment:
I don't consider them methods of the instance, but rather transformation functions that can be called on a port, much in the same vein as Extensive. I wouldn't want to confuse instance methods for transformation functions that happen to be stored in the class definition. And theoretically, they can be used as the rule instead of Extensive if for whatever reason the user only wants to split things and never mix, so they follow the same api as Extensive.